pkg/controller/case_test.go

```diff
@@ -0,0 +1,583 @@
+		func(_ *osb.ProvisionRequest) (*osb.ProvisionResponse, error) {
+			numberOfPolls++
+			if numberOfPolls > numberOfErrorResponses {
+				return &osb.ProvisionResponse{}, nil
+			}
+			return nil, osb.HTTPStatusCodeError{
+				StatusCode: http.StatusUnauthorized,
+			}
+		})
+}
+
+func (ct *ControllerTest) SetOSBBindReactionWithHTTPError(code int) {
+	ct.fakeOSBClient.BindReaction = &fakeosb.BindReaction{
+		Error: osb.HTTPStatusCodeError{
+			StatusCode: code,
+		},
+	}
+}
+
+func (ct *ControllerTest) spyOSBClientFunc(target v2.CreateFunc) v2.CreateFunc {
+	return func(osbCfg *v2.ClientConfiguration) (v2.Client, error) {
+		ct.osbClientCfg = osbCfg
+		return target(osbCfg)
+	}
+}
+
+func (ct *ControllerTest) createClusterServiceBroker() *v1beta1.ClusterServiceBroker {
+	return &v1beta1.ClusterServiceBroker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testClusterServiceBrokerName,
+		},
+		Spec: v1beta1.ClusterServiceBrokerSpec{
+			CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
+				URL:            "https://broker.example.com",
+				RelistBehavior: v1beta1.ServiceBrokerRelistBehaviorDuration,
+				RelistDuration: &metav1.Duration{Duration: 15 * time.Minute},
+			},
+		},
+	}
+}
+
+func (ct *ControllerTest) CreateSimpleClusterServiceBroker() {
+	ct.scInterface.ClusterServiceBrokers().Create(ct.createClusterServiceBroker())
+}
+
+func (ct *ControllerTest) CreateClusterServiceBrokerWithBasicAuth() {
+	csb := ct.createClusterServiceBroker()
+	csb.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{
+		Basic: &v1beta1.ClusterBasicAuthConfig{
+			SecretRef: &v1beta1.ObjectReference{
+				Name:      authSecretName,
+				Namespace: testNamespace,
+			},
+		},
+	}
+	ct.scInterface.ClusterServiceBrokers().Create(csb)
+}
+
+func (ct *ControllerTest) CreateServiceInstance() {
+	ct.scInterface.ServiceInstances(testNamespace).Create(&v1beta1.ServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testServiceInstanceName,
+		},
+		Spec: v1beta1.ServiceInstanceSpec{
+			PlanReference: v1beta1.PlanReference{
+				ClusterServiceClassExternalName: testClassExternalID,
+				ClusterServicePlanExternalName:  testPlanExternalID,
+			},
+			ExternalID: testExternalID,
+			// Plan and Class refs are added by a Webhook, which is not tested here
+			ClusterServicePlanRef: &v1beta1.ClusterObjectReference{
+				Name: testPlanExternalID,
+			},
+			ClusterServiceClassRef: &v1beta1.ClusterObjectReference{
+				Name: testClassExternalID,
+			},
+		},
+	})
+}
+
+func (ct *ControllerTest) DeleteServiceInstance() {
+	ct.scInterface.ServiceInstances(testNamespace).Delete(testServiceInstanceName, &v1.DeleteOptions{})
+}
+
+func (ct *ControllerTest) CreateBinding() {
+	ct.scInterface.ServiceBindings(testNamespace).Create(&v1beta1.ServiceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:  testNamespace,
+			Name:       testBindingName,
+			Generation: 1,                                         // set by the Webhook
```

Comment:
misleading comment. It's not set by webhook. The main k8s server is doing that.

Reply:
you are right, removed